Move more ivars from WKWebViewConfiguration to API::PageConfiguration

achristensen07 opened the pull request:
#### b2db7454f86da1a03978a6d4460b02c8b9245558
<pre>
Move more ivars from WKWebViewConfiguration to API::PageConfiguration
<a href="https://bugs.webkit.org/show_bug.cgi?id=271405">https://bugs.webkit.org/show_bug.cgi?id=271405</a>
<a href="https://rdar.apple.com/125186260">rdar://125186260</a>

Reviewed by Charlie Wolfe.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::inlineMediaPlaybackRequiresPlaysInlineAttribute const):
(API::PageConfiguration::setInlineMediaPlaybackRequiresPlaysInlineAttribute):
(API::PageConfiguration::allowsInlineMediaPlaybackAfterFullscreen const):
(API::PageConfiguration::setAllowsInlineMediaPlaybackAfterFullscreen):
(API::PageConfiguration::mediaDataLoadsAutomatically const):
(API::PageConfiguration::setMediaDataLoadsAutomatically):
(API::PageConfiguration::dragLiftDelay const):
(API::PageConfiguration::setDragLiftDelay):
(API::PageConfiguration::textInteractionGesturesEnabled const):
(API::PageConfiguration::setTextInteractionGesturesEnabled):
(API::PageConfiguration::longPressActionsEnabled const):
(API::PageConfiguration::setLongPressActionsEnabled):
(API::PageConfiguration::systemPreviewEnabled const):
(API::PageConfiguration::setSystemPreviewEnabled):
(API::PageConfiguration::shouldDecidePolicyBeforeLoadingQuickLookPreview const):
(API::PageConfiguration::setShouldDecidePolicyBeforeLoadingQuickLookPreview):
* Source/WebKit/UIProcess/API/Cocoa/APIPageConfigurationCocoa.mm: Added.
(API::PageConfiguration::Data::defaultShouldDecidePolicyBeforeLoadingQuickLookPreview):
(API::PageConfiguration::Data::defaultDragLiftDelay):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(toDragLiftDelay):
(toWKDragLiftDelay):
(fromWKDragLiftDelay):
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration _inlineMediaPlaybackRequiresPlaysInlineAttribute]):
(-[WKWebViewConfiguration _setInlineMediaPlaybackRequiresPlaysInlineAttribute:]):
(-[WKWebViewConfiguration _allowsInlineMediaPlaybackAfterFullscreen]):
(-[WKWebViewConfiguration _setAllowsInlineMediaPlaybackAfterFullscreen:]):
(-[WKWebViewConfiguration _dragLiftDelay]):
(-[WKWebViewConfiguration _setDragLiftDelay:]):
(-[WKWebViewConfiguration _longPressActionsEnabled]):
(-[WKWebViewConfiguration _setLongPressActionsEnabled:]):
(-[WKWebViewConfiguration _systemPreviewEnabled]):
(-[WKWebViewConfiguration _setSystemPreviewEnabled:]):
(-[WKWebViewConfiguration _shouldDecidePolicyBeforeLoadingQuickLookPreview]):
(-[WKWebViewConfiguration _setShouldDecidePolicyBeforeLoadingQuickLookPreview:]):
(-[WKWebViewConfiguration _mediaDataLoadsAutomatically]):
(-[WKWebViewConfiguration _setMediaDataLoadsAutomatically:]):
(-[WKWebViewConfiguration _textInteractionGesturesEnabled]):
(-[WKWebViewConfiguration _setTextInteractionGesturesEnabled:]):
(defaultShouldDecidePolicyBeforeLoadingQuickLookPreview): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationInternal.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/276517@main">https://commits.webkit.org/276517@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/781cf7bf5e49ed237934f084fc12393aab8dd0ae

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44875 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23974 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47365 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47529 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40880 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28030 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21371 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36842 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45453 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21025 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/38638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17927 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18437 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2922 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41103 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40075 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49198 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19843 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16388 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/21161 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/21500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6222 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20836 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebKit/UIProcess/API/Cocoa/APIPageConfigurationCocoa.mm:
```diff
@@ -0,0 +1,51 @@
+}
+#endif
+
+} // API
```
I usually see closing namespace comments like `} // namespace API`.

charliewolfe commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
```diff
@@ -137,10 +133,6 @@ - (instancetype)init
     _allowsPictureInPictureMediaPlayback = YES;
 #endif
 
-    _allowsInlineMediaPlayback = !PAL::currentUserInterfaceIdiomIsSmallScreen();
```
`#import <pal/system/ios/UserInterfaceIdiom.h>` can probably be removed from this file now.

achristensen07 committed:
Move more ivars from WKWebViewConfiguration to API::PageConfiguration
https://bugs.webkit.org/show_bug.cgi?id=271405
rdar://125186260

Reviewed by Charlie Wolfe.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::inlineMediaPlaybackRequiresPlaysInlineAttribute const):
(API::PageConfiguration::setInlineMediaPlaybackRequiresPlaysInlineAttribute):
(API::PageConfiguration::allowsInlineMediaPlaybackAfterFullscreen const):
(API::PageConfiguration::setAllowsInlineMediaPlaybackAfterFullscreen):
(API::PageConfiguration::mediaDataLoadsAutomatically const):
(API::PageConfiguration::setMediaDataLoadsAutomatically):
(API::PageConfiguration::dragLiftDelay const):
(API::PageConfiguration::setDragLiftDelay):
(API::PageConfiguration::textInteractionGesturesEnabled const):
(API::PageConfiguration::setTextInteractionGesturesEnabled):
(API::PageConfiguration::longPressActionsEnabled const):
(API::PageConfiguration::setLongPressActionsEnabled):
(API::PageConfiguration::systemPreviewEnabled const):
(API::PageConfiguration::setSystemPreviewEnabled):
(API::PageConfiguration::shouldDecidePolicyBeforeLoadingQuickLookPreview const):
(API::PageConfiguration::setShouldDecidePolicyBeforeLoadingQuickLookPreview):
* Source/WebKit/UIProcess/API/Cocoa/APIPageConfigurationCocoa.mm: Added.
(API::PageConfiguration::Data::defaultShouldDecidePolicyBeforeLoadingQuickLookPreview):
(API::PageConfiguration::Data::defaultDragLiftDelay):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(toDragLiftDelay):
(toWKDragLiftDelay):
(fromWKDragLiftDelay):
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration _inlineMediaPlaybackRequiresPlaysInlineAttribute]):
(-[WKWebViewConfiguration _setInlineMediaPlaybackRequiresPlaysInlineAttribute:]):
(-[WKWebViewConfiguration _allowsInlineMediaPlaybackAfterFullscreen]):
(-[WKWebViewConfiguration _setAllowsInlineMediaPlaybackAfterFullscreen:]):
(-[WKWebViewConfiguration _dragLiftDelay]):
(-[WKWebViewConfiguration _setDragLiftDelay:]):
(-[WKWebViewConfiguration _longPressActionsEnabled]):
(-[WKWebViewConfiguration _setLongPressActionsEnabled:]):
(-[WKWebViewConfiguration _systemPreviewEnabled]):
(-[WKWebViewConfiguration _setSystemPreviewEnabled:]):
(-[WKWebViewConfiguration _shouldDecidePolicyBeforeLoadingQuickLookPreview]):
(-[WKWebViewConfiguration _setShouldDecidePolicyBeforeLoadingQuickLookPreview:]):
(-[WKWebViewConfiguration _mediaDataLoadsAutomatically]):
(-[WKWebViewConfiguration _setMediaDataLoadsAutomatically:]):
(-[WKWebViewConfiguration _textInteractionGesturesEnabled]):
(-[WKWebViewConfiguration _setTextInteractionGesturesEnabled:]):
(defaultShouldDecidePolicyBeforeLoadingQuickLookPreview): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationInternal.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/276517@main
